enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,191 @@
+
+The following steps should be done to enable unicast for `ingress-vip`:
+1. Update the Keepalived conf template files for both [control plane nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/master/00-master/baremetal/files/baremetal-keepalived-keepalived.yaml#L49-#L73) and [compute nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/worker/00-worker/baremetal/files/baremetal-keepalived-keepalived.yaml) to include the unicast details.
+2. The keepalived-monitor container should be updated to [retrieve](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/config/node.go#L219-#L244) the nodes (both compute and control) IP addresses from the kube-apiserver and render them to Keepalived config.
+
+### Risks and Mitigations
+
+It is important to not destabilize our existing installation method, that uses
+multicast successfully. Full parity is expected before landing.
+
+Since upgrade may trigger a Keepalived configuration migration process (see upgrade section below), it's important to add upgrade coverage for bare metal IPI in CI.
+
+## Design Details
+
+### Test Plan
+
+Because of the fundamental inclusion of this in Baremetal IPI installations,
+any breakage by CI before landing can probably be assumed to be caused by
+this work.
+
+Take down nodes both in active as in passive mode and observe the VIP move correctly and without significantly increasing the current expected downtime in VIP changes.
+
+Force a disaster recovery scenario and see that we can recover the cluster following the usual procedure.
+
+Since the baremetal-runtimecfg is used by all on-prem platforms, we should verify that Ovirt, vSphere, and OpenStack still successfully deployed in Keepalived multicast mode.
+
+### Upgrade / Downgrade Strategy
+
+In case that the Keepalived config file was already generated by the previous SW version, Keepalived will continue to run in a mode determined in accordance with the existing file.
+If the cluster admin wants to change the Keepalived mode post upgrade process, he should trigger a configuration migration process.
+
+The next steps illustrate Keepalived flow in case of upgrade/downgrade:
+1. If config file exists determined_mode=mode determined from the file else determined_mode=mode specified in MCO templates
+2. Run Keepalived in determined_mode.
+
+#### Configuration migration process
+
+Since multicast and unicast are considered as separate Keepalived domains and will not interoperate, the updated configuration should simultaneously reload in all the relevant Keepalived instances. This option is feasible since the nodes run NTP.
+
+The next steps describe how the Keepalived mode should be changed in the cluster:
+
+1. Cluster admin should create a MachineConfig CR, this CR will generate a new file in all nodes.
```

Comment:
I think we should have this be an automated process that no admin needs to deal with.  Any thoughts on how that might be done?

Reply:
Yep, we can use MCO node's annotation as @celebdor suggested.
I'll update this section

Reply:
Done